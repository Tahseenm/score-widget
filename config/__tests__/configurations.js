'use strict'

const babelConfig   = require('../babel')
const browsersList  = require('../browsersList')
const postCSSConfig = require('../postcss')


const PRODUCTION_ENV  = 'production'
const DEVELOPMENT_ENV = 'development'
const INVALID_ENV     = 'foobarbaz'


/**
 Babel Conifguration Tests
 */
describe('Babel Configuration', () => {
  test('Invalid Enviroment', () => {
    const getConfig = () => babelConfig(INVALID_ENV)
    expect(getConfig).toThrow()
  })

  test('Production Enviroment', () => {
    const config = babelConfig(PRODUCTION_ENV)
    expect(config).toMatchSnapshot()
  })

  test('Development Enviroment', () => {
    const config = babelConfig(DEVELOPMENT_ENV)
    expect(config).toMatchSnapshot()
  })
})


/**
 Browser Lists Configuration Tests
 */
describe('BrowsersList Configuration', () => {
  test('Invalid Enviroment', () => {
    const getConfig = () => browsersList(INVALID_ENV)
    expect(getConfig).toThrow()
  })

  test('Production Enviroment', () => {
    const config = browsersList(PRODUCTION_ENV)
    expect(config).toMatchSnapshot()
  })

  test('Development Enviroment', () => {
    const config = browsersList(DEVELOPMENT_ENV)
    expect(config).toMatchSnapshot()
  })
})


/**
 PostCSS Configuration Tests
 */
describe('PostCSS Configuration', () => {
  test('Invalid Enviroment', () => {
    const getConfig = () => postCSSConfig(INVALID_ENV)
    expect(getConfig).toThrow()
  })

  test('Production Enviroment', () => {
    const config = postCSSConfig(PRODUCTION_ENV)
    expect(config).toMatchSnapshot()
  })

  test('Development Enviroment', () => {
    const config = postCSSConfig(DEVELOPMENT_ENV)
    expect(config).toMatchSnapshot()
  })
})
