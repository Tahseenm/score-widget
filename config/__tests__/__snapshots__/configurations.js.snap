// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel Configuration Development Enviroment 1`] = `
Object {
  "plugins": Array [
    "syntax-dynamic-import",
    "transform-object-rest-spread",
    "transform-class-properties",
  ],
  "presets": Array [
    "react",
    Array [
      "env",
      Object {
        "ignoreBrowserslistConfig": true,
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "last 2 Chrome versions",
            "last 2 Safari major versions",
            "last 2 Edge versions",
            "last 2 Firefox versions",
          ],
        },
      },
    ],
  ],
}
`;

exports[`Babel Configuration Production Enviroment 1`] = `
Object {
  "plugins": Array [
    "syntax-dynamic-import",
    "transform-object-rest-spread",
    "transform-class-properties",
  ],
  "presets": Array [
    "react",
    Array [
      "env",
      Object {
        "ignoreBrowserslistConfig": true,
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "> 1%",
            "not IE < 11",
          ],
        },
      },
    ],
  ],
}
`;

exports[`BrowsersList Configuration Development Enviroment 1`] = `
Array [
  "last 2 Chrome versions",
  "last 2 Safari major versions",
  "last 2 Edge versions",
  "last 2 Firefox versions",
]
`;

exports[`BrowsersList Configuration Production Enviroment 1`] = `
Array [
  "> 1%",
  "not IE < 11",
]
`;

exports[`PostCSS Configuration Development Enviroment 1`] = `
Object {
  "plugins": Array [
    [Function],
    [Function],
  ],
}
`;

exports[`PostCSS Configuration Production Enviroment 1`] = `
Object {
  "plugins": Array [
    [Function],
    [Function],
  ],
}
`;
